nativeconverter/zstdchunked/zstdchunked.go

```diff
@@ -101,6 +111,14 @@ func LayerConvertFuncWithCompressionLevel(compressionLevel zstd.EncoderLevel, op
 		uncompressedDesc := &desc
 		// We need to uncompress the archive first
 		if !uncompress.IsUncompressedType(desc.MediaType) {
+			mu.Lock()
+			if _, ok := uncompressMap[desc.Digest]; !ok {
+				uncompressMap[desc.Digest] = &unMap{}
+			}
+			uncompressMap[desc.Digest].mu.Lock()
+			mu.Unlock()
+			defer uncompressMap[desc.Digest].mu.Unlock()
+
```

Comment:
Sorry for slow review. I think we can just remove ` cs.Delete(ctx, uncompressedDesc.Digest)` in this function. containerd  garbage collects that blob when nobody refers to that.

Reply:
Hey @ktock 

I don't know...
Is `uncompress.LayerConvertFunc` going to be safe if used concurrently on the same resource?

Anyhow, I am fine with this getting fixed any way, so, feel free to adopt / change it / carry it over.